### PR TITLE
Integrate Firebase Auth UI

### PR DESCRIPTION
**Pull request by Alton09**
## Description

Replaced the internal login UI with the firebase auth UI SDK. This reduces the amount of code we need to support and removes the need for the EmailAuthProvider and GoogleAuthProvider. This will also lead the way to remove the need of the RxJava Observables for the login method within the Authenticator, and the RxJava dependency all together in favor of coroutines. A future PR will encompass that work because some refactoring will need to be done for the community authenticator logic.

## Validation

- Manually validated the community login and firebase auth flows.
- Passed CI.

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in ```versionName``` under `app/build.gradle`
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

**Comment by Alton09** [comment]
One community unit test failure. I'll fix it on monday.

**Comment by Alton09** [comment]
Alright fixed the test. CI is passing now ✅

File: app/src/main/java/com/twilio/video/app/ui/login/FirebaseAuthActivity.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                // Successfully signed in
+                val user = FirebaseAuth.getInstance().currentUser
+                Timber.d("Logged in was successful!!")
+                // ...
+            } else {
+                Timber.d("Logged in was not successful :(")
+                // Sign in failed. If response is null the user canceled the
+                // sign-in flow using the back button. Otherwise check
+                // response.getError().getErrorCode() and handle the error.
+                // ...
```
Comment (kbagchiGWC) [inline app/src/main/java/com/twilio/video/app/ui/login/FirebaseAuthActivity.kt]: Is this a TODO? Do you plan to check the error code to do something more specific?

**Review by kbagchiGWC — APPROVED** [review]
Looks great. LGTM!!

File: app/src/main/java/com/twilio/video/app/ui/login/FirebaseAuthActivity.kt (hunk 1 above)
Comment (Alton09) [reply app/src/main/java/com/twilio/video/app/ui/login/FirebaseAuthActivity.kt]: Ah yeah sorry for the confusion. This class no longer exists. I was going back and forth on where to put the FirebaseUI builder and ended up deciding to just include it in the LoginActivity class.

**Comment by Alton09** [comment]
Fixed red background on email button and blue "Having trouble sigining in" text view.

**Review by kbagchiGWC — APPROVED** [review]

**Pull request merged**